account_vendor_bank_account_default_purchase/models/purchase_order.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+from odoo import api, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def _get_default_supplier_partner_bank(self, partner):
+        if not partner.has_default_bank_id:
+            return False
+        elif partner.default_bank_id:
+            return partner.default_bank_id.id
+        else:
+            return super()._get_default_supplier_partner_bank(partner)
+
+    @api.depends("partner_id", "company_id")
```

Comment:
I don't think you need to include the depends decorator when no new dependencies are added.

Reply:
I include it for readability, but i can remove it

Reply:
Up to you really